File: output/objc/AddressBuilder.m
#import "AddressBuilder.h"

@implementation AddressBuilder
{
    NSNumber* _postalCode;
    NSString* _streetAddress;
    NumberEnum _number;
    BOOL _valid;
}

- (Address *)build
{
    return [Address addressWithPostalCode:_postalCode streetAddress:_streetAddress number:_number valid:_valid];
}

#pragma mark - Initializers

+ (instancetype)builder
{
    return [self new];
}

+ (instancetype)builderFromAddress:(Address *)existingAddress
{
    AddressBuilder* builder = [self builder];
    builder = [builder withPostalCode:existingAddress.postalCode];
    builder = [builder withStreetAddress:existingAddress.streetAddress];
    builder = [builder withNumber:existingAddress.number];
    builder = [builder withValid:existingAddress.isValid];
    return builder;
}

#pragma mark - Property setters

- (instancetype)withPostalCode:(NSNumber*)postalCode
{
    _postalCode = [postalCode copy];
    return self;
}

- (instancetype)withStreetAddress:(NSString*)streetAddress
{
    _streetAddress = [streetAddress copy];
    return self;
}

- (instancetype)withNumber:(NumberEnum)number
{
    _number = number;
    return self;
}

- (instancetype)withValid:(BOOL)valid
{
    _valid = valid;
    return self;
}

@end

File: output/swift/AddressBuilder.swift
import Foundation

struct AddressBuilder {
	var postalCode: Int
	var streetAddress: String
	var number: Int
    
    func build () -> Address {
        return Address(postalCode: postalCode, streetAddress: streetAddress, number: number)
    }
    
    //MARK: Initializers
    
    init(existingAddress: Address) {
        postalCode = existingAddress.postalCode
        streetAddress = existingAddress.streetAddress
        number = existingAddress.number
    }
    
    //MARK: Property setters
    
    mutating func withPostalCode(_ postalCode: Int) -> AddressBuilder {
        self.postalCode = postalCode
        return self
    }

    mutating func withStreetAddress(_ streetAddress: String) -> AddressBuilder {
        self.streetAddress = streetAddress
        return self
    }

    mutating func withNumber(_ number: Int) -> AddressBuilder {
        self.number = number
        return self
    }
}
